File: go/labels/type.go
package labels

type Label struct {
	name      string
	value     any
}

func NewLabel(name string, value any) Label {
	return Label{name: name, value: value}
}

func name[T any](name string) func(T) Label {
	return func(value T) Label {
		return NewLabel(name, value)
	}
}

func (l Label) Name() string {
	return l.name
}

func (l Label) Value() any {
	return l.value
}

File: python/labels/type.py
from typing import Any

class Label():
    def __init__(self, name: str, value: Any):
        self._name = name
        self._value = value

    @property
    def name(self) -> str:
        return self._name

    @property
    def value(self) -> Any:
        return self._value 
    
    @staticmethod
    def name(name: str):
        return lambda value: Label(name, value)
    
